pkg/adapters/aiven.go

```
@@ -0,0 +1,542 @@
+	dbtuneConfig.BindEnv("service_name", "DBT_AIVEN_SERVICE_NAME")
+
+	var aivenConfig adeptersinterfaces.AivenConfig
+	err := dbtuneConfig.Unmarshal(&aivenConfig)
+	if err != nil {
+		return nil, fmt.Errorf("unable to decode into struct: %v", err)
+	}
+
+	// Validate required configuration
+	err = utils.ValidateStruct(&aivenConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	// Create Aiven client
+	aivenClient, err := aiven.NewClient(aiven.TokenOpt(aivenConfig.APIToken))
+	if err != nil {
+		return nil, fmt.Errorf("failed to create Aiven client: %v", err)
+	}
+
+	// Create default PostgreSQL adapter as base
+	defaultAdapter, err := CreateDefaultPostgreSQLAdapter()
+	if err != nil {
+		return nil, fmt.Errorf("failed to create base PostgreSQL adapter: %v", err)
+	}
+
+	// Create adapter
+	adapter := &AivenPostgreSQLAdapter{
+		DefaultPostgreSQLAdapter: *defaultAdapter,
+		aivenConfig:              aivenConfig,
+		aivenClient:              aivenClient,
+		state: &adeptersinterfaces.AivenState{
+			LastAppliedConfig: time.Time{},
+		},
+	}
+
+	// Initialize collectors
+	adapter.MetricsState = agent.MetricsState{
+		Collectors: AivenCollectors(adapter),
+		Cache:      agent.Caches{},
+		Mutex:      &sync.Mutex{},
+	}
+
+	return adapter, nil
+}
+
+func (adapter *AivenPostgreSQLAdapter) PGDriver() *pgPool.Pool {
+	return adapter.pgDriver
+}
+
+// GetSystemInfo returns system information for the Aiven PostgreSQL service
+func (adapter *AivenPostgreSQLAdapter) GetSystemInfo() ([]utils.FlatValue, error) {
+	adapter.Logger().Info("Collecting Aiven system info")
+
+	var systemInfo []utils.FlatValue
+
+	// Get service information from Aiven API
+	ctx := context.Background()
+	service, err := adapter.aivenClient.ServiceGet(
+		ctx,
+		adapter.aivenConfig.ProjectName,
+		adapter.aivenConfig.ServiceName,
+		[2]string{"include_secrets", "false"},
+	)
+
+	if err != nil {
+		adapter.Logger().Errorf("failed to get service info: %v", err)
```

this will get logged twice, you can just return the error
```suggestion
```